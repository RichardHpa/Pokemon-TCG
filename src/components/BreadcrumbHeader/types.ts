interface Breadcrumb {
  label: string
  path?: string
}

export interface BreadcrumbHeaderProps {
  title: string
  breadcrumbs?: Breadcrumb[]
}
