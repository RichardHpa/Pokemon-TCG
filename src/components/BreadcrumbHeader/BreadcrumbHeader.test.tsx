import { render, screen, within } from 'tests/renderHelpers'
import { BreadcrumbHeader } from './BreadcrumbHeader'

describe('BreadcrumbHeader', () => {
  test('renders an BreadcrumbHeader', () => {
    render(<BreadcrumbHeader title='Page Title' />)
    expect(screen.getByRole('heading', { name: 'Page Title', level: 3 })).toBeInTheDocument()
  })

  test('renders breadcrumbs', () => {
    const { container } = render(
      <BreadcrumbHeader
        title='Page Title'
        breadcrumbs={[
          {
            label: 'crumb1',
            path: '/crumb1',
          },
          {
            label: 'crumb2',
          },
        ]}
      />,
    )
    expect(screen.getByRole('heading', { name: 'Page Title', level: 3 })).toBeInTheDocument()
    const listItems = screen.getAllByRole('listitem')
    expect(listItems).toHaveLength(3)

    expect(within(listItems[0]).getByLabelText('Return home')).toBeInTheDocument()
    expect(within(listItems[1]).getByRole('link', { name: 'crumb1' })).toBeInTheDocument()
    expect(within(listItems[2]).getByText('crumb2')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
