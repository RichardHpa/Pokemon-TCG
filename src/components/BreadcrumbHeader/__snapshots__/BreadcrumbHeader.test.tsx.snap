// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbHeader renders breadcrumbs 1`] = `
<div>
  <div
    class="mb-4"
  >
    <nav
      aria-label="Breadcrumb"
      class="flex mb-4"
    >
      <ol
        class="inline-flex items-center space-x-1 md:space-x-3"
      >
        <li
          class="inline-flex items-center"
        >
          <a
            aria-label="Return home"
            class="inline-flex items-center text-sm font-medium text-gray-700 hover:text-blue-600 dark:text-gray-400 dark:hover:text-white"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
              />
            </svg>
          </a>
        </li>
        <li>
          <div
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-6 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                fill-rule="evenodd"
              />
            </svg>
            <a
              class="ml-1 text-xs font-medium text-gray-700 hover:text-blue-600  dark:text-gray-400 dark:hover:text-white"
              href="/crumb1"
            >
              crumb1
            </a>
          </div>
        </li>
        <li>
          <div
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-6 h-4 text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              aria-current="page"
              class="ml-1 text-xs font-medium text-gray-700 dark:text-gray-400"
            >
              crumb2
            </span>
          </div>
        </li>
      </ol>
    </nav>
    <h3
      class="font-bold tracking-tight text-gray-900 dark:text-white mb-2 text-3xl"
    >
      Page Title
    </h3>
  </div>
</div>
`;
